core/src/main/java/org/opensearch/sql/expression/operator/predicate/RegexMatch.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.operator.predicate;
+
+import java.util.regex.PatternSyntaxException;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.model.ExprValueUtils;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ExpressionNodeVisitor;
+import org.opensearch.sql.expression.env.Environment;
+import org.opensearch.sql.expression.parse.RegexCommonUtils;
+
+/**
+ * Expression for regex matching using Java's built-in regex engine. Supports standard Java regex
+ * features including named groups, lookahead/lookbehind, backreferences, and inline flags. Uses
+ * find() for partial matching semantics.
+ */
+@ToString
+@EqualsAndHashCode
+public class RegexMatch implements Expression {
```

Comment:
Is this for V2 engine?

Reply:
Good catch, and this was an left over from my poc branch which I was trying to support it for legacy engine. Since now, we are targeting in calcite only, so I have removed the above.